compiler/rustc_span/src/def_id.rs

```diff
@@ -279,6 +279,7 @@ impl DefId {
     }
 
     #[inline]
+    #[track_caller]
     pub fn expect_local(self) -> LocalDefId {
         self.as_local().unwrap_or_else(|| panic!("DefId::expect_local: `{:?}` isn't local", self))
```

Comment:
This doesn't work because there's a closure in the middle which doesn't track caller.
So either `#[track_caller]` need to be applied to the closure too, or `match` need to be used instead of `unwrap_or_else`.

Reply:
Didn't know, I'll address it. Thanks!

Reply:
Does the same apply to `expect_non_local`?
https://github.com/rust-lang/rust/blob/9714e139ffb25f3b96b28110ae81677c8c1a7cf2/compiler/rustc_hir/src/def.rs#L633-L636

Reply:
Yes, `fn expect_non_local` should have the same issue.